loadtest/loadtest.go

```diff
@@ -0,0 +1,328 @@
+package loadtest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/helm/pkg/helm"
+)
+
+type Config struct {
+	Clients        *Clients
+	GuestFramework *framework.Guest
+	Logger         micrologger.Logger
+
+	AuthToken    string
+	ClusterID    string
+	CommonDomain string
+}
+
+type LoadTest struct {
+	clients        *Clients
+	guestFramework *framework.Guest
+	logger         micrologger.Logger
+
+	authToken    string
+	clusterID    string
+	commonDomain string
+}
+
+func New(config Config) (*LoadTest, error) {
+	if config.Clients.ControlPlaneHelmClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Clients.ControlPlaneHelmClient must not be empty", config)
+	}
+	if config.Clients.ControlPlaneK8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Clients.ControlPlaneK8sClient must not be empty", config)
+	}
+	if config.GuestFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestFramework must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.AuthToken == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AuthToken must not be empty", config)
+	}
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+	if config.CommonDomain == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CommonDomain must not be empty", config)
+	}
+
+	s := &LoadTest{
+		clients:        config.Clients,
+		guestFramework: config.GuestFramework,
+		logger:         config.Logger,
+
+		authToken:    config.AuthToken,
+		clusterID:    config.ClusterID,
+		commonDomain: config.CommonDomain,
+	}
+
+	return s, nil
+}
+
+func (l *LoadTest) Test(ctx context.Context) error {
```

Comment:
Personal opinion, for me `Run` sounds better

Reply:
Uncertain. I'm OK with the name and its the convention we use in e2etests.